view/src/components/DiffViewWrapper.jsx

```diff
@@ -0,0 +1,236 @@
+      if (oldChangeKey === newChangeKey) return [oldChangeKey];
+      return [oldChangeKey, newChangeKey];
+    }
+  }
+  return [];
+}
+
+function getSyntaxHighlightedTokens(hunks) {
+  const options = {
+    highlight: true,
+    refractor,
+    language: 'c',
+  };
+  return tokenize(hunks, options);
+}
+
+/**
+ * A component for visualisation of source code of two versions of function.
+ * For the function are shown lines of code:
+ * - which contains differences,
+ * - first and last line of the function.
+ * If the function is calling another function (linesToShow is set) than
+ * - if the function is short (<= MAX_LINES_THRESHOLD) than code of the whole function is shown,
+ * - else the line with the call is shown
+ *   and certain amount (LINES_OF_CONTEXT) of lines above and below the call.
+ * For the lines which are not shown are rendered buttons that allow user to show these lines
+ * by clicking on the button.
+ * @param {Object} props
+ * @param {string} props.oldCode - Source code of file containing old version of the function.
+ * @param {string} props.diff - Output of unified diff of old and new source code.
+ * @param {number} props.oldStart - Line where the old version of the function starts.
+ * @param {number} props.newStart - Line where the new version of the function starts.
+ * @param {number} props.oldEnd - Line where the old version of the function ends.
+ * @param {Boolean} [props.showDiff=true] - True to color differences in function.
+ * @param {Number} [props.linesToShow=null] - Lines (old, new) of function which
+ *                                            should be shown and highlighted.
+ */
+export default function DiffViewWrapper({
+  oldCode,
+  diff,
+  oldStart,
+  newStart,
+  oldEnd,
+  showDiff = true,
+  linesToShow = null,
+}) {
+  /* getting object which Diff component needs */
+  const file = useMemo(() => {
+    if (diff !== '') {
+      const [fileObject] = parseDiff(diff);
+      return fileObject;
+    }
+    return createFileStructure(oldCode, oldStart, newStart);
+  }, [diff, oldCode, oldStart, newStart]);
+    /* hunks of differences/code which will be shown */
+  const [hunks, setHunks] = useState(null);
+  // function for adding lines of old code form startLine to endLine
+  // which should be shown
+  const expandHunks = useCallback(
+    (startLine, endLine = startLine) => {
+      setHunks((oldHunks) => expandFromRawCode(oldHunks, oldCode, startLine, endLine + 1));
+    },
+    [oldCode],
+  );
+
+  /**
+   * Setting hunks which will be shown.
+   * Shown are lines:
+   *  - with differences between files/functions,
+   *  - first and last line of function,
+   * If the linesToShow is set (the function is calling another function) than
+   * other lines of function are shown:
+   *  - if the function is short than code of whole function is shown,
+   *  - if the function is longer than the lines of code are shown (linesToShow)
+   *    and few lines above and bellow it.
+   */
```

Comment:
I don't think that it's necessary to duplicate the description. Maybe just add a comment pointing to the component description comment.

Reply:
Thanks, done.